### PR TITLE
PUD-742  BE: add explicit persistence of no known vulnerability & contraband

### PR DESCRIPTION
**Pull request by AndyNewton**
BE: add explicit persistence for both:
* no known vulnerability/diversity issue and 
* contraband intent

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallSummaryGenerator.kt [hunk 1]
```diff
@@ -56,9 +56,10 @@ class RecallSummaryGenerator(
       setVariable("releaseDate", recallSummaryContext.lastReleaseDate.asStandardDateFormat())
       setVariable("furtherCharge", recallSummaryContext.reasonsForRecall.isFurtherCharge())
       setVariable("policeSpoc", recallSummaryContext.localPoliceForce)
-      setVariable("vulnerabilityDetail", ValueOrNone(recallSummaryContext.vulnerabilityDiversityDetail))
-      setVariable("contraband", YesOrNo(recallSummaryContext.contrabandDetail.hasContrabandDetail()))
+      setVariable("contraband", YesOrNo(recallSummaryContext.contraband ?: false))
       setVariable("contrabandDetail", recallSummaryContext.contrabandDetail)
+      setVariable("vulnerabilityDiversity", YesOrNo(recallSummaryContext.vulnerabilityDiversity ?: false))
```
Comment (AndyNewton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallSummaryGenerator.kt:61]: Full disclosure ... I believe this line is unnecessary and should be removed.

Comment (AndyNewton) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallSummaryGenerator.kt]: ^^^ Removed in update.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/lettertoprison/LetterToPrisonCustodyOfficeGenerator.kt [hunk 2]
```diff
@@ -38,10 +38,10 @@ class LetterToPrisonCustodyOfficeGenerator(
         setVariable("recallLengthDays", RecallLengthDescription(this.recallLength).numberOfDays())
         setVariable("hasAdditionalLicenceConditions", this.additionalLicenceConditions)
         setVariable("additionalLicenceConditionsDetail", this.additionalLicenceConditionsDetail)
-        setVariable("hasVulnerabilities", this.vulnerabilityDiversityDetail?.isNotEmpty())
-        setVariable("vulnerabilityDiversityDetail", this.vulnerabilityDiversityDetail)
-        setVariable("hasContraband", this.contrabandDetail?.isNotEmpty())
+        setVariable("hasContraband", this.contraband ?: false)
```
Comment (philsegal) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/lettertoprison/LetterToPrisonCustodyOfficeGenerator.kt:41]: Should we be defaulting this? Feels like we should force it (!!) as its an expected value.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/lettertoprison/LetterToPrisonCustodyOfficeGenerator.kt [hunk 3]
```diff
@@ -38,10 +38,10 @@ class LetterToPrisonCustodyOfficeGenerator(
...
         setVariable("contrabandDetail", this.contrabandDetail)
+        setVariable("hasVulnerabilities", this.vulnerabilityDiversity ?: false)
```
Comment (philsegal) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/lettertoprison/LetterToPrisonCustodyOfficeGenerator.kt:43]: and here

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallSummaryGenerator.kt [hunk 4]
```diff
@@ -56,9 +56,9 @@ class RecallSummaryGenerator(
       setVariable("lastReleaseDate", recallSummaryContext.lastReleaseDate.asStandardDateFormat())
       setVariable("furtherCharge", recallSummaryContext.reasonsForRecall.isFurtherCharge())
       setVariable("localPoliceForce", recallSummaryContext.localPoliceForce)
-      setVariable("vulnerabilityDiversityDetail", ValueOrNone(recallSummaryContext.vulnerabilityDiversityDetail))
-      setVariable("hasContrabandDetail", YesOrNo(recallSummaryContext.contrabandDetail.hasContrabandDetail()))
+      setVariable("hasContrabandDetail", YesOrNo(recallSummaryContext.contraband ?: false))
```
Comment (philsegal) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallSummaryGenerator.kt:59]: and here

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/lettertoprison/LetterToPrisonCustodyOfficeGenerator.kt (hunk 2 above)
Comment (AndyNewton) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/lettertoprison/LetterToPrisonCustodyOfficeGenerator.kt]: Done.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/lettertoprison/LetterToPrisonCustodyOfficeGenerator.kt (hunk 3 above)
Comment (AndyNewton) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/lettertoprison/LetterToPrisonCustodyOfficeGenerator.kt]: Done.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallSummaryGenerator.kt (hunk 4 above)
Comment (AndyNewton) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallSummaryGenerator.kt]: Done.

**Review by philsegal — APPROVED** [review]

File: src/main/resources/db/migration/V43__Add_vulnerability_and_contraband_booleans.sql [hunk 5]
```diff
@@ -0,0 +1,7 @@
+ALTER TABLE recall
```
Comment (AndyNewton) [inline src/main/resources/db/migration/V43__Add_vulnerability_and_contraband_booleans.sql:1]: Improved script per discussion with devs.

**Review by mikedunfordEE — APPROVED** [review]

**Pull request merged**